Python/TelemetrySystem/telemetry.py

```diff
@@ -1,25 +1,25 @@
-from client import TelemetryClient
+from client import AbstractClient
+
 
 class TelemetryDiagnostics:
-    DiagnosticChannelConnectionString = "*111#"
+    DIAGNOSTIC_CHANNEL_CONNECTION_STRING = "*111#"
 
-    def __init__(self):
-        self._telemetry_client = TelemetryClient()
+    def __init__(self, client: AbstractClient, retries: int = 3):
+        self.retries = retries
+        self._telemetry_client = client
         self.diagnostic_info = ""
 
-    def check_transmission(self):
+    def check_transmission(self, client_message: str):
         self.diagnostic_info = ""
         self._telemetry_client.disconnect()
 
-        retryLeft = 3
-        while ((not self._telemetry_client.online_status) and retryLeft > 0):
-            self._telemetry_client.connect(TelemetryDiagnostics.DiagnosticChannelConnectionString)
-            retryLeft -= 1
+        retry_left = self.retries
+        while not self._telemetry_client.online_status and retry_left > 0:
+            self._telemetry_client.connect(self.DIAGNOSTIC_CHANNEL_CONNECTION_STRING)
+            retry_left -= 1
 
         if not self._telemetry_client.online_status:
             raise Exception("Unable to connect.")
 
-        self._telemetry_client.send(TelemetryClient.DIAGNOSTIC_MESSAGE)
+        self._telemetry_client.send(client_message)
         self.diagnostic_info = self._telemetry_client.receive()
```

Comment:
and perhaps, send method should also return the response? Do we really need to split the logic into two methods?

Reply:
I thought we dont need to change the logic in the client class.

Reply:
we could change anything we want =)

Reply:
I have updated the logic